torchtitan/checkpoint.py

```diff
@@ -59,12 +61,30 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         set_optimizer_state_dict(self.model, self.optim, optim_state_dict=state_dict)
 
 
+class DataLoaderWrapper:
+    def __init__(self, dataloader: DataLoader) -> None:
+        self.dataloader = dataloader
+        self.rank_id = (
+            dist.get_rank() if (dist.is_available() and dist.is_initialized()) else 0
+        )
+
+    def state_dict(self) -> Dict[str, Any]:
+        if isinstance(self.dataloader, StatefulDataLoader):
+            return {self.rank_id: self.dataloader.state_dict()}
+        return {}
+
+    def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
+        if isinstance(self.dataloader, StatefulDataLoader):
+            self.dataloader.load_state_dict(state_dict[self.rank_id])
```

Comment:
This seems to break because `rank_id` gets cast to a string at some point. Should maybe cast it to a string manually in both functions?